### ISSUE TITLE
Error installing module in Docker containers

### ISSUE THREAD

**Issue by edsonmarquezani** [firebase/firebase-js-sdk#1395]
I'm getting the following error when installing firebase.
```
npm ERR! path /home/jenkins/workspace/er-frontend_frontend-ci_dev-T3NMY3HDJF7V26MAHVO4AFXGHZ6BCH4JZUWQHNS6LQE2ET2BJH7A/commons/node_modules/@firebase/polyfill/node_modules/core-js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/home/jenkins/workspace/er-frontend_frontend-ci_dev-T3NMY3HDJF7V26MAHVO4AFXGHZ6BCH4JZUWQHNS6LQE2ET2BJH7A/commons/node_modules/@firebase/polyfill/node_modules/core-js' -> '/home/jenkins/workspace/er-frontend_frontend-ci_dev-T3NMY3HDJF7V26MAHVO4AFXGHZ6BCH4JZUWQHNS6LQE2ET2BJH7A/commons/node_modules/@firebase/polyfill/node_modules/.core-js.DELETE'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```
This seems to be related to this bug -> https://github.com/npm/npm/issues/9863, which affects a lot of projects, actually. Mainly, overlay filesystem doesn't support the rename syscall.

Anyway, I can't `npm install` any project which has Firebase as dependency when running inside containers. Really problematic. Would it be possible to implement a workaround for this?

**Comment by Feiyang1** [comment]
The issue you linked seems to be fixed. Did you have this issue before or after upgrading the firebase version. I doubt it has anything to do with firebase though. 
Can you please try to delete the node_modules folder in your projects and do `npm cache clean --force` and make a clean install ?

**Comment by edsonmarquezani** [comment]
The problem is definitely not related to cache, once I build in ephemeral Jenkins slaves (in containers, off course). So I'm always running a fresh install.

Regarding the time it began, I have this problem since I tried to build it inside containers for the very first time. It took some time for me to realize it was related to Docker Filesystem, actually.

I've tried with at least 2 different Docker version:
- 17.03.2-ce (`--storage-driver=overlay`)
- 17.12.1-ce (`--storage-driver=overlay2`)

I haven't tried other storage drivers yet.
